extensions/configuration-editing/schemas/devContainer.schema.src.json

```diff
@@ -253,51 +253,101 @@
 				"onCreateCommand": {
 					"type": [
 						"string",
-						"array"
+						"array",
+						"object"
 					],
-					"description": "A command to run when creating the container. This command is run after \"initializeCommand\" and before \"updateContentCommand\". If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell.",
+					"description": "A command to run when creating the container. This command is run after \"initializeCommand\" and before \"updateContentCommand\". If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell. Alternatively an object with the property keys as labels and the values as either strings or arrays of strings can be used to configure multiple commands.",
 					"items": {
 						"type": "string"
+					},
+					"additionalProperties": {
+						"type": [
+							"string",
+							"array"
+						],
+						"items": {
+							"type": "string"
+						}
 					}
 				},
 				"updateContentCommand": {
 					"type": [
```

Comment:
@aeschli You recommended using "oneOf" in a similar case. Does this apply here too? Would "string" and "array" go in one clause and "object" in another? What is the advantage of "oneOf"? Thanks!

Reply:
Chatted with @aeschli and we can include this as-is. oneOf is easier to support on the tooling side, but I can revisit when we find this is important. Thanks.